src/main/scala/com/ing/wbaa/gargoyle/sts/oauth/OAuth2Authorization.scala

```diff
@@ -39,16 +39,24 @@ class OAuth2Authorization(oAuth2tokenVerifier: OAuth2TokenVerifier) {
   }
 
   /**
-   * get the token from header or cookie or parameters
+   * get the token from header or cookie or parameters or body
    *
    * @return the authorization token
    */
   private def bearerToken: Directive1[Option[String]] =
     for {
```

Comment:
I'm not too happy with this for-comprehension. Can this be made clearer somehow? There's a lot going on in there…

Reply:
split to more functions